server/sonar-server/src/main/java/org/sonar/server/authentication/JwtTokenUpdater.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.authentication;
+
+import io.jsonwebtoken.Claims;
+import java.util.Date;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.lang.time.DateUtils;
+import org.sonar.api.platform.Server;
+import org.sonar.api.server.ServerSide;
+import org.sonar.api.utils.System2;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.user.UserDto;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.util.Arrays.asList;
+
+@ServerSide
+public class JwtTokenUpdater {
+
+  private static final Logger LOG = Loggers.get(GenerateJwtTokenFilter.class);
+
+  private static final String JWT_COOKIE = "JWT-SESSION";
+
+  // This timeout is used to disconnect the user we he has not browse any page for a while
+  private static final int SESSION_TIMEOUT_IN_SECONDS = 20 * 24 * 60 * 60;
+
+  // This refresh time is used to generate a new session
+  private static final int SESSION_REFRESH_IN_MINUTES = 5;
+
+  private static final String RAILS_USER_ID_SESSION = "user_id";
+
+  private final System2 system2;
+  private final DbClient dbClient;
+  private final Server server;
+  private final JwtToken jwtToken;
+
+  public JwtTokenUpdater(System2 system2, DbClient dbClient, Server server, JwtToken jwtToken) {
+    this.jwtToken = jwtToken;
+    this.server = server;
+    this.dbClient = dbClient;
+    this.system2 = system2;
+  }
+
+  void createNewJwtToken(String userLogin, HttpServletResponse response) {
```

Comment:
In a first implementation I was injecting the UserSession, but then I realize it was not possible because of IdentityProvider : they're using this method to create a token, but UserSession is not feed at that time when an IdentityProvider is used.
We can have too method but I would prefer to wait for Java to create the UserSession (because then IdentityProvider will create the UserSession)